js/single-docset-search.tsx

```diff
@@ -6,9 +6,57 @@ import cx from "classnames";
 
 import elasticlunr from "elasticlunr";
 
-elasticlunr.tokenizer.setSeperator(/[\s-.>=+\/]+/);
+const SEARCHSET_VERSION = 3;
+
+function tokenize(str?: string): string[] {
+  if (!arguments.length || str === null || str === undefined) return [];
+  // split on conventional whitespace
+  const candidateTokens = str.toString().trim().toLowerCase().split(/\s+/);
+  const resultingTokens: string[] = [];
+  candidateTokens.forEach(function (candidate: string) {
+    // ignore if all non-alphanumeric and greater >= 7 characters
+    // this keeps things like *, <, >, +, ->> but turfs things like ===============
+    if (
+      !/^[^a-z0-9]{7,}$/.test(candidate) &&
+      // ignore stand-alone comment characters
+      !/^;+$/.test(candidate)
+    ) {
+      // strip leading and trailing periods and commas
+      // this gets rid of normal punctuation
+      // we leave in ! and ? because they can be interesting in var names
+      const token = candidate.replace(/^[.,]+|[.,]+$/g, "");
+      if (token.length > 0) {
+        resultingTokens.push(token);
+      }
+    }
+  });
+  return resultingTokens;
+}
+
+function subTokenize(tokens: string[]): string[] {
+  const resultingSubTokens: string[] = [];
+  tokens.forEach(function (token: string) {
+    // break down into subTokens, if appropriate, for example
+    // clojure.core.test would break down to clojure core test
+    if (/[^a-z0-9]/.test(token)) {
+      const subTokens = token.split(/[^a-z0-9]+/);
+      subTokens.forEach(function (subToken: string) {
+        if (subToken.length > 0) {
+          resultingSubTokens.push(subToken);
+        }
+      });
+    }
```

Comment:
Nice one here, optimizing for the common case of tokens without non-alphas in them. At first glance I thought the first test was wasteful but no this is actually saving creating a lot of arrays.